### DISCUSSION TITLE
any good documentation for a beginner ?

### DISCUSSION THREAD

**Discussion by teto** [nvim-orgmode/orgmode#182] [Q&A]
hi,

super happy to see an orgmode for neovim, I've heard so many good things about it that I am eager to try it but I've no idea where to start with. For instance I've created TODO tasks and was expecting them to show up in the calendar but they dont ? 
These TODOs are listed in /home/teto/orgmode/refile.org instead.

Could the README recommand an adapted tutorial ? is there a chat channel to ask questions live ? like a subchannel in the neovim matrix space ?

The plugin looks polished and I can't wait to tap into the full potential of the plugin !

**Comment by levouh** [comment]
The calendar is only for choosing dates related to items; if you want to see your "schedule" you are looking for the _Agenda_ view (opened with `<leader>oa` by default).

_Orgmode_ is a general enough "task management" tool that I'm not sure a generalized tutorial would be incredibly useful. Perhaps individual tutorials for each of the separate features might be useful, then the user can integrate them into their workflow as needed.

Not sure where something like that would start, but the general ask around "I have no clue how _Orgmode_ works" is a seemingly common problem @kristijanhusak. I've never really found a good "meta-level" discussion about _Orgmode_ (there are some for things like _GTD_, e.g. [here](http://members.optusnet.com.au/~charles57/GTD/gtd_workflow.html) or [here](https://whhone.com/posts/org-mode-task-management/)), and most examples (e.g. [here](https://orgmode.org/worg/org-tutorials/orgtutorial_dto.html) and the previous links) are _Emacs_-specific (as you would expect).

Also a _Discord_ or _Matrix_ channel would be awesome :wink:

**Reply by teto** [reply]
thanks for the answer, if I open `t List of all TODO entries ` I see my TODO entries. 
If I look at section 3 of https://whhone.com/posts/org-mode-task-management/ , he sees his TODO items directly in the weekly calendar. Why don't I ? Right now, I've in my refile.org 
```
* TODO post on the blog
  sdadasda
  [2021-12-12 Sun]
  DEADLINE: <2020-11-07 Sat> SCHEDULED: <2020-10-31 Sat>
* TODO support mptcp versions 0 and 1
  adasdasdas
  [2021-12-12 Sun]
```

As for a room, @clason has been trying to regroup plugins in a neovim subspace for ease of discoveribility in https://app.element.io/#/room/#neovim_space:matrix.org so if there is interest, we could create one there.

**Reply by kristijanhusak** [reply]
I never really planned to have a "teaching section" on how to use Orgmode. I believe that if someone knows to use Orgmode in Emacs, they shouldn't have any problems figuring things out here. 
Do you know how Emacs is handling this? I know they have a subreddit, but I wouldn't go that far. Regarding Matrix or Discord, I'm not really using any of those, but feel free to set up a space there if it will help someone.
I guess having dedicated github pages with some "Getting started" for now would be helpful.

I'll create an issue for this with a "Help needed" label, and I hope someone will jump into doing that. I'm trying to focus my free time on adding new features.

@teto you can look into this video to get a little idea how orgmode works https://www.youtube.com/watch?v=nsv33iOnH34 . It's a different plugin, but most of the ideas from there are similar.

**Reply by kristijanhusak** [reply]
I added this link to the [Getting started](https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#getting-started-with-orgmode) section in docs for now.
![meme](https://i.kym-cdn.com/entries/icons/original/000/028/021/work.jpg)

**Reply by halfdan** [reply]
For me the problem is that I'm not an emacs user and don't want to be. Having some sort of documented "happy path" for basic note taking and task management would allow me to ramp up much faster when this plugin is my very first interaction with orgmode.

**Reply by jgollenz** [reply]
@halfdan it seems we're having a hard time doing this properly. Do you have any examples of other repos/tools that have provided you with a useful happy path from which we can get some inspiration?

**Comment by jgollenz** [comment]
@kristijanhusak let's please talk about this. I don't think we have given documentation enough attention. Even if this were a perfect port of orgmode, if people don't know how to use it, there is little point in it. Some thoughts I have:

 
- The github pages were a good start, but I hardly remember that they even exist. Are they in sync with the current state of the repo? How do we make sure they stay in sync?
- What about the 'Wiki' section of a github repo? In my opinion it helps to keep the whole thing a bit more compact (in the sense of 'everything is at a single place'), although it gives you less flexibility on how exactly you want to lay out your docs. In any case, having both github pages and a wiki is probably not a good idea.
- A 'getting started' section has been missing for far too long. 
- The 'DOCS' link in the README links to `DOCS.md`. The link to the 'Getting started' section is not easy to find when just skimming through the README. This might leave users with the impression that there is no 'getting started' section (let's ignore the fact that this is currently true for a second)
- The headline levels of the README also seem to be out of sync, the links on top don't cover items on the same level. The ones with the same level are 
	- Installation
	- Setup
	- Plugins
	- Development
	- Plans (has no link on top)
	- Kudos

but we also have
	- Troubleshoot
	- Gifs
	- Docs	
	- Tree-sitter info
linked at the top, giving the impression that all of them are on the same level

Since multiple people have requested it, I made a matrix room: https://matrix.to/#/#neovim-orgmode:matrix.org

**Reply by kristijanhusak** [reply]
I fully agree that documentation is lacking, and it needs to be improved a lot. Unfortunately it's also time consuming. When I catch time I usually try to address bugs and do some minor refactorings, while poorly maintaining DOCS.md that is used both as markdown documentation and Vim help documentation. If someone would take some part in this, I would help reviewing it and adding own suggestions. 

Regarding your questions:

> The github pages were a good start, but I hardly remember that they even exist. Are they in sync with the current state of the repo? How do we make sure they stay in sync?

It's on the separate repo in the organization https://github.com/nvim-orgmode/nvim-orgmode.github.io. They were not touched since they were created by @gerazov .


> What about the 'Wiki' section of a github repo? In my opinion it helps to keep the whole thing a bit more compact (in the sense of 'everything is at a single place'), although it gives you less flexibility on how exactly you want to lay out your docs. In any case, having both github pages and a wiki is probably not a good idea.

We could have wiki for some getting started and some tips n' tricks, or just a more detailed version of readme file. I don't think we should go more than that if we will have a documentation repo.
On the other side, we could archive the documentation repo, and use only Wiki, which would simplify contributions.

Generally both README and DOCS needs some attention for those things, like TOC and moving things around to make sense.
As I said above, we use DOCS.md for both repo and Vim documentation. I tried to avoid duplicating work with that, but I don't think we will be able to do that. At least 2 docs sources will need to be maintained to have a decent documentation:
* Vim help
* Other docs (which is now only DOCS.md + some parts of README), either the docs repo that we have or Wiki

I don't want @gerazov effort to be thrown away, but I think maybe using Wiki would be much simpler, at least from the contribution side.

Unfortunately I never got an answer to the question above(https://github.com/nvim-orgmode/orgmode/discussions/182#discussioncomment-1794082) how Emacs people learn Orgmode? Of course there are bunch of tutorials online and blog posts, but that requires effort outside of the Orgmode itself. 
Like, do they just go to https://orgmode.org/manual/ and start reading the docs and using it? I'm genuinely curious, and would love to get an answer from someone who is using the Emacs Orgmode. 
Regarding my learning path, I actually never actively used Emacs Orgmode until I started working on this package. I used vim-dotoo, and then started developing this. Of course I installed Emacs to test things how they do it, and I read a lot of the manual, but never really used it day to day. That's why I probably have this somewhat unreal expectation that other people should do the same thing, which is wrong.

To summarize:

* README needs to be improved with TOC and some reorganization
* DOCS.md could use some attention, but it does the trick considering it's used for 2 "sources"
* More detailed documentation (Getting Started, tips n' tricks, etc.) is definitely needed, but I'm not sure if we should use separate repo or Wiki. For now I would go with Wiki since it's much easier to contribute, but I'm open to suggestions.

**Reply by clason** [reply]
> Since multiple people have requested it, I made a matrix room: https://matrix.to/#/#neovim-orgmode:matrix.org 

Want me to add this to the ["Neovim Plugins" subspace](https://app.element.io/#/room/#neovim-plugins:matrix.org) on Matrix?

**Reply by kristijanhusak** [reply]
Yeah please do that.

**Reply by clason** [reply]
Done.

**Reply by jgollenz** [reply]
> it's also time consuming.
 
Absolutely. And it's by far not as fun as adding features :grin: 

> If someone would take some part in this, I would help reviewing it and adding own suggestions.

Consider me in :+1: 

> They were not touched since they were created by @gerazov

So they are probably not in sync. I agree with you that wikis make contributions much easier and they also make the whole project more consolidated. Indeed, it is a shame if we have to throw away the work @gerazov did. Everything that is in a github wiki is part of a separate repo, in our case it is `nvim-orgmode/orgmode.wiki.git`, where every page is stored as a `.md` file. Perhaps there is a way how we can take those, process them accordingly so ReadTheDocs can work with them and continue to host them with github pages. Perferable automated at every change to the wiki.

**Reply by kristijanhusak** [reply]
> Everything that is in a github wiki is part of a separate repo, in our case it is nvim-orgmode/orgmode.wiki.git

TIL. That's very neat.

> Perhaps there is a way how we can take those, process them accordingly so ReadTheDocs can work with them and continue to host them with github pages. Perferable automated at every change to the wiki.

This would be awesome, and seems ~plausible~ possible.

**Reply by jgollenz** [reply]
@kristijanhusak if you create a wiki section for the repo I'll start with adding a 'getting started' tutorial :slightly_smiling_face:

**Reply by kristijanhusak** [reply]
@jgollenz should be unlocked now. I thought it was already open.

**Reply by jgollenz** [reply]
I'd love some help :heart: it's probably easiest to discuss it on the [matrix channel](https://matrix.to/#/#neovim-orgmode:matrix.org)

[a comment by an automated account is not shown]

**Comment by jgollenz** [comment]
@teto we now have a [Getting Started](https://github.com/nvim-orgmode/orgmode/wiki/Getting-Started) section in our wiki. Please have a look at it and tell us what you think :+1:

**Reply by teto** [reply]
thanks a lot, that's what I was looking for. Some examples of https://github.com/nvim-orgmode/orgmode/wiki/Getting-Started#links seem to lack either a `[` or `]`.
I believe org has many features so might be worth using subtrees in the outline, the most interesting aspect to me for instance is "literate programming" aka "babel" (I think ?!)with `#+BEGIN_SRC` markers. I dont think it's quite available yet but really looking forward to that.

I am also curious as to what "refiling" means.

**Reply by jgollenz** [reply]
Thanks for catching the missing brackets, should be fixed now. 

I think literate programming is a large enough topic for it's own page, but a short example in the 'First Steps' sounds like a good idea :+1:

@teto @aareman btw, you both (and everybody else) should be able to just edit the wiki pages. Whenever you spot errors or want to improve something, feel free to do so :slightly_smiling_face:

**Comment by aareman** [comment]
Refiling means moving a heading tree to a new location. Like refiling from one filing cabinet to another. You can also refile it under another heading in the same or different files.
________________________________

**Comment by ryflamm** [comment]
I think it would be helpful to have an example org file/directory people can download and poke around. And yes, I know I can find emacs org files out in the wild... but those may have things that nvim-orgmode doesn't support (yet) or has implemented a bit differently. For instance, best I can tell nvim implementation doesn't support top of file properties like `#+STARTUP: indent align overview` to set file specific behaviors, tags, etc.
